### PR TITLE
Eslint, audit fixes, circleci v2, sonar fixes

### LINKED ISSUE #105: CircleCI v2
Right now we are using the v1, but it's gonna be disposed on March 15

Steps to do:
- add .circleci/config.yml
- make sure all the tests run on circleci
- make sure sonar code analysis is sent

### PR DESCRIPTION
**Pull request by Spring3**
resolves #105

### CODE REVIEW

**Comment by dialonce** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [consumer.js#L33](https://github.com/dial-once/node-bunnymq/blob/3d4cae9ba2363d6610932865a9fc24361e39b4a9/src/modules/consumer.js#L33): Define a constant instead of duplicating this literal 4 times. [rule](https://sonar.dialonce.net/coding_rules#rule_key=javascript%3AS1192)
1. ![CRITICAL][CRITICAL] [producer.js#L59](https://github.com/dial-once/node-bunnymq/blob/3d4cae9ba2363d6610932865a9fc24361e39b4a9/src/modules/producer.js#L59): Define a constant instead of duplicating this literal 3 times. [rule](https://sonar.dialonce.net/coding_rules#rule_key=javascript%3AS1192)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

File: .npmignore [hunk 1]
```diff
@@ -26,6 +26,8 @@ build/Release
 # https://www.npmjs.org/doc/misc/npm-faq.html#should-i-check-my-node_modules-folder-into-git
 node_modules
 .env
+# Because we want peopel to receive patch fixes of dependency modules during the npm i
```
Comment (mrister) [inline .npmignore]: minor typo, peopel

**Comment by dialonce** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [consumer.js#L33](https://github.com/dial-once/node-bunnymq/blob/5e8dd63d0ea885c828d5022b2129c8e62808fc01/src/modules/consumer.js#L33): Define a constant instead of duplicating this literal 4 times. [rule](https://sonar.dialonce.net/coding_rules#rule_key=javascript%3AS1192)
1. ![CRITICAL][CRITICAL] [producer.js#L59](https://github.com/dial-once/node-bunnymq/blob/5e8dd63d0ea885c828d5022b2129c8e62808fc01/src/modules/producer.js#L59): Define a constant instead of duplicating this literal 3 times. [rule](https://sonar.dialonce.net/coding_rules#rule_key=javascript%3AS1192)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

**Review by mrister — APPROVED** [review]
Lookin' good!

File: src/modules/producer.js [hunk 2]
```diff
@@ -56,7 +58,7 @@ class Producer {
       }
 
       // if we found one, we execute the callback and delete it because it will never be received again anyway
-      this._connection.config.transport.info('bmq:producer', `[${queue}] < answer`);
+      this._connection.config.transport.info(loggerAlias, `[${queue}] < answer`);
```
Comment (shamil) [inline src/modules/producer.js:60]: line 53, should also have `loggerAlias` IMO

Comment (Spring3) [reply src/modules/producer.js:60]: I agree.

**Review by ykhatal — APPROVED** [review]

**Comment by shamil** [comment]
Hi @Spring3 can we expect a release soon?

**Comment by Spring3** [comment]
@shamil we try to release projects at the beginning of the week

**Pull request merged**